src/NUnitEngine/nunit.engine.core/RuntimeFramework.cs

```diff
@@ -201,6 +201,8 @@ private Version GetClrVersionForFramework(Version frameworkVersion)
                             return new Version(5, 0, 1);
                         case 6:
                             return new Version(6, 0, 0);
+                        case 7:
+                            return new Version(7, 0, 0);
```

Comment:
@CharliePoole : I was curious if this would be a problem once .NET 8 (and future versions of .NET) get released...
/cc: @OsirisTerje

Reply:
Yes, it would, at least for anyone continuing to use 3.15.2 at that point. However, subsequent changes, to be released in 3.16, eliminated throwing of an exception when the runtime is not recognized. In addition, version 4.0 code eliminates this call entirely, dropping the need for the CLR version entirely. Both of the above are superior, in my opinion, to the often suggested option of using `<MajorVersion>.0.0` because that involves predicting Microsoft's future actions.

Reply:
That makes a lot of sense. Glad the newer versions have this fixed up.

Reply:
@CharliePoole I'm hitting this again with .NET 8 preview installed. How do I get VS/dotnet test to use 3.16?

Reply:
That's based on the version of the NUnit VS adapter you are using. Check with @OsirisTerje on that project.

Reply:
@madelson We need a new version of the engine, 3.16.3.  It should be released any day now, and then version 4.40 of the adapter will have this embedded.  It will be released immediately after the 3.16.3